src/xrpld/app/tx/detail/CreateOffer.cpp

```diff
@@ -1259,6 +1367,24 @@ CreateOffer::applyGuts(Sandbox& sb, Sandbox& sbCancel)
         sleOffer->setFlag(lsfPassive);
     if (bSell)
         sleOffer->setFlag(lsfSell);
+    if (domainID)
+        sleOffer->setFieldH256(sfDomainID, *domainID);
+
+    // if it's a hybrid offer, set hybrid flag, and create an open dir
+    if (bHybrid)
+    {
+        std::function<void(SLE::ref, bool)> setDirFunc = setBookDir;
+        if (auto const res = applyHybrid(
+                sb,
+                sleOffer,
+                offer_index,
+                saTakerPays,
+                saTakerGets,
+                setDirFunc);
```

Comment:
just `setBookDir `should work, no?

Reply:
a1171876e26dc5df9a4cfb5f12b05a685ae569ab